### PR TITLE
polish printing, and fix printing for kwargs, etc. add deps on MLStyle

### LINKED ISSUE #3: Support varargs and kwargs
Currently both varargs and kwargs cause errors as they’re not correctly recognized by the macro. 
```julia
julia> @lambda(;x=1.0 -> x+ 1)
ERROR: LoadError: "Must be called on a Lambda expression"
Stacktrace:
 [1] @λ(::LineNumberNode, ::Module, ::Any) at /home/mason/.julia/packages/LegibleLambdas/N2VRk/src/LegibleLambdas.jl:28
in expression starting at /home/mason/.julia/packages/LegibleLambdas/N2VRk/src/LegibleLambdas.jl:90

julia> @lambda(x=1.0 -> x+ 1)
ERROR: LoadError: "Must be called on a Lambda expression"
Stacktrace:
 [1] @λ(::LineNumberNode, ::Module, ::Any) at /home/mason/.julia/packages/LegibleLambdas/N2VRk/src/LegibleLambdas.jl:28
in expression starting at /home/mason/.julia/packages/LegibleLambdas/N2VRk/src/LegibleLambdas.jl:90

```

### PR DESCRIPTION
**Pull request by Roger-luo**
Should fix #3 and this provides the functionality to extend and define custom printing. I'll add some more tests, or someone could help me add more tests.

### CODE REVIEW

**Comment by Roger-luo** [comment]
Maybe add a `Project.toml`? so we could develop this by `git clone` and use a local environment. And I guess it's time to remove the dependency on `MacroTools` later, we don't really need that `postwalk` function. 

PS. @MasonProtter could you add me to collaborator?

**Comment by MasonProtter** [comment]
Looking good. I added you as a collaborator.

File: src/LegibleLambdas.jl [hunk 1]
```diff
@@ -17,100 +17,20 @@ julia> @λ(x -> g(x)/3)
...
-
-    function Base.show(io::IO, ::MIME"text/plain", f::LegibleLambda)
-        print(io, to_string(f))
-    end
-
-end
-
 macro lambda(ex)
     return :(@λ($ex))
 end
```
Comment (MasonProtter) [inline src/LegibleLambdas.jl]: I'm not certain, but I think this should read
```julia
macro lambda(ex)
    return esc(:(@λ($ex)))
end
```
instead. This is subtle but it causes problems with things like 
```julia
julia> @lambda((x; y=1) -> x + y)
ERROR: syntax: "LegibleLambdas.x" is not a valid function argument name
```
whereas 
```julia
macro lambda2(ex)
        return esc(:(@λ($ex)))
end

julia> @lambda2((x; y=1) -> x + y)
(begin
          x
          y = 1
      end -> x + y)
```

**Comment by MasonProtter** [comment]
Okay, I think I've got it working properly now. Keyword arguments are still ugly but not the end of the world. 

```julia
julia> @λ (x; y=1) -> x^2 + y^2
(begin
          x
          y = 1
      end -> x ^ 2 + y ^ 2)
```

@Roger-luo Should we try and fix that now, or in a later PR?

**Comment by MasonProtter** [comment]
Okay, I've changed it so that kwargs now essentially print like varargs
```julia
julia> @λ (x; y=1) -> x^2 + y^2
((x, y = 1) -> x ^ 2 + y ^ 2)
```
This still has the problem of being ambitious with the printing for varargs, so the next step should be to  detect where the kwarg list starts and place a `;` before it instead of a `,` in the string representation.

File: src/LegibleLambdas.jl (hunk 1 above)
Comment (Roger-luo) [reply src/LegibleLambdas.jl]: Ah, yeah, good gotcha

**Comment by Roger-luo** [comment]
looks good to merge. Keywords are great now! I guess we only need polish some small details in the future, but already looks nice now.

I polished the project configuration, so we could start use Pkg3 instead of `REQUIRE` from now on. It will simplify a lot of things, e.g you can directly `git clone` the repo and `cd` to the folder, then just type `julia --project`, julia will run it locally, instead of a global environment messed up with other packages.

**Pull request merged**